### PR TITLE
METRON-295: Script parsing bolt

### PR DESCRIPTION
**Pull request by rkarthik29**
I have added support for creating parsers using js, python or groovy and using them in metron. This is for Jira Metron-295

### CODE REVIEW

**Comment by cestella** [comment]
Hi Karthik.  Can you close this and reopen with the title: "METRON-295: Script parsing bolt"?

**Comment by rkarthik29** [comment]
i changed the title

**Comment by ottobackwards** [comment]
don't forget the reopen part

**Comment by ottobackwards** [comment]
This is interesting.  I would think we would want some tooling like https://github.com/YelpArchive/pyleus around building a jar that contains all the python dependencies for a script for people to do non-trivial things in python.

I am not sure how you would do the same for javascript or groovy.

File: metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/ScriptParser.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+		        if (fieldValue != null) {
+		          message.put(timeField, toEpoch(fieldValue));
+		        }
+		      }
+		      if (timestampField != null) {
+		        message.put(Constants.Fields.TIMESTAMP.getName(), formatTimestamp(message.get(timestampField)));
+		      }
+		      messages.add(message);
+		      if (LOG.isDebugEnabled()) {
+		        LOG.debug("Grok parser parsed message: " + message);
```
Comment (james-sirota) [inline metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/ScriptParser.java]: Looks like cut and paste error from the Grok code

File: metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/ScriptParser.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+		      }
+		}catch (Exception e) {
+		    LOG.error(e.getMessage(), e);
+		    throw new IllegalStateException("Grok parser Error: " + e.getMessage() + " on " + originalMessage , e);
```
Comment (james-sirota) [inline metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/ScriptParser.java]: Here is another one

**Pull request closed without merging**